From da70856a2f7ba2f29eb5dcd1ce71deec1a59f225 Mon Sep 17 00:00:00 2001
From: Zhen Han <zhen.han@intel.com>
Date: Fri, 06 Sep 2019 14:22:40 +0800
Subject: [PATCH] Telephony: return ModemActivityInfo when defaultPhone is null

It will be timeout in getting ModemActivityInfo
when no modem or defaultphone is present in the Andorid platforms
(e.g.: Android TV or IVI). Specifically, it would wastes 2s and
result in CTS test failure finally in waiting the result of
this message requeset and result in CTS test failure.

This patch returns ModemActivityInfo with all fields set to 0
when defaultPhone is null.

Change-Id: I4210a90e1fe6a7e0e087caf69831813dd2adf434
Tracked-On:
Signed-off-by: Zhen Han <zhen.han@intel.com>
---

diff --git a/src/com/android/phone/PhoneInterfaceManager.java b/src/com/android/phone/PhoneInterfaceManager.java
index d0ed9bd..ae39e2f 100755
--- a/src/com/android/phone/PhoneInterfaceManager.java
+++ b/src/com/android/phone/PhoneInterfaceManager.java
@@ -844,6 +844,9 @@
                     onCompleted = obtainMessage(EVENT_GET_MODEM_ACTIVITY_INFO_DONE, request);
                     if (defaultPhone != null) {
                         defaultPhone.getModemActivityInfo(onCompleted, request.workSource);
+                    } else {
+                        request.result = new ModemActivityInfo(0, 0, 0, null, 0, 0);
+                        notifyRequester(request);
                     }
                     break;
 
